### ISSUE TITLE
Is this possible?

### ISSUE THREAD

**Issue by resistorsoftware** [Shopify/shopify_app#1699]
Say I have webhooks set in the Admin to 2023-04. Say I submitted to Shopify that this App does not bother with any customer information, but they never did anything about that. Say I ask for scope write_orders. So now I install this public App already installed in App Store in a store, and request a webhook for orders/paid. 

I then query this store and see only one webhook ever installs, the app/uninstalled one that goes along with orders/paid. 

Is it possible that because for whatever reason Shopify never ticked off the box on the customer privacy boxes, that a webhook accessing orders would simply not be installed? 

If this is true, and I think it might be, would it not be nice to at least log that? So the developer knows?

**Comment by key88sf** [comment]
If you have a public app that is already installed by customers, and you deploy a code change which *adds* new webhooks to the list managed by the ShopifyApp config, do those new webhooks get automatically installed for the *existing* customers using the app, even if they don't uninstall/reinstall the app itself?

**Comment by resistorsoftware** [comment]
At the time of an old install being re-authenticated, the webhooks are audited and missing one are added. The caveat being, as I am pointing out here, that with PCD issues, you can have those installs fail for not meeting PCD criteria and have no clue that that is happening, as this App does not log these failures.

**Comment by key88sf** [comment]
> At the time of an old install being re-authenticated, the webhooks are audited and missing one are added.

When does the old install get re-authenticated? I'm trying to figure out if existing app users need to do something special for the new webhooks to be picked up.

**Comment by resistorsoftware** [comment]
My assumption, shallow as it is, it is the same as when you change scopes, and the App checks for those changes and then flashes up a Modal at the merchant to accept those changes. The Webhook registrations are checked against ones present, and then missing ones are created. If not, that makes this App less nice.

**Comment by key88sf** [comment]
Yeah that is what I'm seeing from local testing as well. That presents a problem though. Consider this:
* You have a public app with existing users.
* You add a new feature that requires a new webhook (but NO change to scopes).
* You update the public app with the new code, which exposes this new feature, but the new required webhook doesn't get registered.

Am I missing something here?

**Comment by resistorsoftware** [comment]
They get registered regardless of scope change. Problem is, if that registration fails, you don't see that in your logs.
